persistence-utils-demo/pom.xml

```diff
@@ -10,6 +10,7 @@
       <groupId>org.hibernate</groupId>
       <artifactId>hibernate-jpamodelgen</artifactId>
       <version>5.4.21.Final</version>
+      <scope>provided</scope>
```

Comment:
looks as if this 'provided' doesn't work as we expected it to do -> the build pipeline fails ... so I can't accept the PR as is

try to work with scope 'runtime' ... or if that leads to the same results, use exclusions in order to minimize the impact of this dependency:
```
<exclusions><exclusion><groupId>*</groupId><artifactId>*</artifactId></exclusion>
</exclusions>
```